homework/list.js

```diff
@@ -1,79 +1,99 @@
 'use strict';
+(function(exports) {
 
-(function() {
+  /**
+  * Create a instance of the ListManager.
+  *
+  * @constructor
+  */
+	var ListManager = function() {
+		/** @private */ this._listNoteContent = [];
+		/** @private */ this._wrapper = document.querySelector('#note-list-wrapper');
+		document.addEventListener('DOMContentLoaded', this.start.bind(this));
+	}
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+	ListManager.prototype = {
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+		/** Initialize the listManager. */
+		start() {
+			this.fetchList()
+				.then(this.updateList.bind(this))
+				.then(this.drawList.bind(this))
+				.then(this.preloadFirstNote.bind(this))
+				.catch(function(error) {
+					console.error(error);
+				});
+			window.addEventListener('click', this.onNoteOpen.bind(this));
+		},
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+		/**
+    * Get List form localhost
+    *
+    * @return {Promise} A start point of the work flow.
+    */
+		fetchList(afterFetch) {
```

Comment:
Since you have already "Promisified" the function, the callback argument here is unnecessary.